Mathlib/RingTheory/FiniteLength.lean

```diff
@@ -76,3 +75,28 @@ theorem isFiniteLength_iff_exists_compositionSeries :
   ⟨fun h ↦ have ⟨_, _⟩ := isFiniteLength_iff_isNoetherian_isArtinian.mp h
     exists_compositionSeries_of_isNoetherian_isArtinian R M,
     isFiniteLength_of_exists_compositionSeries⟩
+
+theorem IsSemisimpleModule.finite_tfae [IsSemisimpleModule R M] :
+    List.TFAE [Module.Finite R M, IsNoetherian R M, IsArtinian R M, IsFiniteLength R M,
+      ∃ s : Set (Submodule R M), s.Finite ∧ CompleteLattice.SetIndependent s ∧
+        sSup s = ⊤ ∧ ∀ m ∈ s, IsSimpleModule R m] := by
+  rw [isFiniteLength_iff_isNoetherian_isArtinian]
+  obtain ⟨s, hs⟩ := IsSemisimpleModule.exists_setIndependent_sSup_simples_eq_top R M
+  tfae_have 1 ↔ 2 := ⟨fun _ ↦ inferInstance, fun _ ↦ inferInstance⟩
+  tfae_have 2 → 5 := fun _ ↦ ⟨s, CompleteLattice.WellFoundedGT.finite_of_setIndependent hs.1, hs⟩
+  tfae_have 3 → 5 := fun _ ↦ ⟨s, CompleteLattice.WellFoundedLT.finite_of_setIndependent hs.1, hs⟩
+  tfae_have 5 → 4 := fun ⟨s, fin, _, sSup_eq_top, simple⟩ ↦ by
+    rw [← isNoetherian_top_iff, ← Submodule.topEquiv.isArtinian_iff,
+      ← sSup_eq_top, sSup_eq_iSup, ← iSup_subtype'']
+    rw [SetCoe.forall'] at simple
+    have := fin.to_subtype
+    exact ⟨isNoetherian_iSup, isArtinian_iSup⟩
+  tfae_have 4 → 2 := And.left
+  tfae_have 4 → 3 := And.right
+  tfae_finish
+
+instance [IsSemisimpleModule R M] [Module.Finite R M] : IsArtinian R M :=
+  (IsSemisimpleModule.finite_tfae.out 0 2).mp ‹_›
+
+example [IsSemisimpleRing R] : IsNoetherianRing R := inferInstance
```

Comment:
These should be tests importing Mathlib if you want to assured that downstream users can synthesize them. 

If you want to document to the user that they can be found, you can record a comment.

Reply:
Where are these tests usually located? I think these are key results that follow from instance just proven so they naturally fit here, and I would hesitate to move them to another file. (The Noetherian example is already available in the SimpleModule file, but I put it here for coherence.) I can definitely make them comments, but I think making them examples ensures that no one (doing refactors etc.) would accidentally break them.

Reply:
Working in this file doesn't guarantee that they work downstream of the file. To avoid future breakage, these should be in a file in the `tests` folder with the `inferInstance`s and `import Mathlib`. This assures that if someone changes any part of the library and breaks the current synthesis path they will be responsible for restoring it.

Reply:
Let me make them comments then. They'll probably break only when the reducibility of IsNoetherianRing/IsArtinianRing, but in that case I'd expect a bunch of other things would also break.